subsys/net/openthread/Kconfig.defconfig

```diff
@@ -25,7 +25,6 @@ config OPENTHREAD_LIBRARY_AVAILABLE
 		    OPENTHREAD_NORDIC_LIBRARY_FTD || OPENTHREAD_NORDIC_LIBRARY_MTD
 	depends on !OPENTHREAD_COPROCESSOR
 	depends on SOC_NRF52840 || SOC_NRF5340_CPUAPP
-	depends on FPU
```

Comment:
Hmm, what would be impact for Matter?
https://github.com/nrfconnect/sdk-connectedhomeip/blob/a2850b23aa044227a245142647265b6ad9cfe1ea/config/nrfconnect/chip-module/Kconfig.defaults#L43-L44

@Damian-Nordic do you have any thoughts?

Reply:
It might be as library path is calculated depending on FPU.  You can see that paths have changed on our test run: https://github.com/nrfconnect/sdk-nrfxlib/pull/1388/files

Reply:
So we're not going to provide hard-float libraries anymore? Then shouldn't it depend on `!FPU`?
Matter shouldn't need FPU directly (although CI will tell the trough). I hope our customers don't as well :)

Reply:
But setting `FPU` for Matter should be removed to keep Matter using the libraries.

Reply:
> So we're not going to provide hard-float libraries anymore? Then shouldn't it depend on `!FPU`? 
To be honest I'm not even sure why this check was even there. Imo it is redundant as if FPU is set incorrectly, lib path calculation will fail anyways.

Reply:
This was to default to using libraries if libraries are available, or to sources otherwise. So that a user doesn't get a failure if they want to build OT sample with/without FPU. But if you think we should make user set `OPENTHREAD_SOURCES=y` when they choose incompatible configuration, that's probably also OK.